eth/query/block.go

```diff
@@ -180,7 +180,10 @@ func GetTxObjectFromBlockResult(
 				}
 				contractAddress = eth.EncPtrAddress(resp.Contract)
 				if len(respData.TxHash) > 0 {
-					txObj.Hash = eth.EncBytes(respData.TxHash)
+					// Check duplicate EVM tx hash before using it
+					if evmAuxStore.IsDupEVMTxHash(respData.TxHash) {
+						txObj.Hash = eth.EncBytes(respData.TxHash)
```

Comment:
The idea was to replace the EVM tx hash by the TM tx hash, this doesn't do that.

Reply:
Fixed